src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -186,280 +214,319 @@ def slice_info_for_literal(self, length, prefix="") -> RawSliceInfo:
-                                open_ = m_open.group(1)
-                                close_ = m_close.group(1)
-                                alternate_code = (
-                                    f"\0{unique_alternate_id} {open_} "
-                                    f"{''.join(trimmed_parts)} {close_}"
-                                )
-                if block_type == "block_start" and trimmed_parts[0] in (
-                    "macro",
-                    "set",
-                ):
-                    # Jinja supports two forms of {% set %}:
-                    # - {% set variable = value %}
-                    # - {% set variable %}value{% endset %}
-                    # https://jinja.palletsprojects.com/en/2.10.x/templates/#block-assignments
-                    # When the second format is used, set the variable 'set_idx'
-                    # to a non-None value. This info is used elsewhere, as
-                    # literals inside a {% set %} block require special handling
-                    # during the trace.
-                    filtered_trimmed_parts = [
-                        p for p in trimmed_parts if not p.isspace()
-                    ]
-                    if (
-                        len(filtered_trimmed_parts) < 3
-                        or filtered_trimmed_parts[2] != "="
-                    ):
-                        set_idx = len(result)
-                elif (
-                    block_type == "block_end"
-                    and set_idx is not None
-                    and (trimmed_parts[0] in ("endmacro", "endset"))
-                ):
-                    # Exiting a {% set %} block. Clear the indicator variable.
-                    set_idx = None
-                m = regex.search(r"\s+$", raw, regex.MULTILINE | regex.DOTALL)
-                if raw.startswith("-") and m:
+                        tag_contents = self.extract_tag_contents(
+                            str_parts, m_close, m_open, str_buff
+                        )
+
+                    if block_type == "block" and tag_contents:
+                        block_type, block_subtype = self.extract_block_type(
+                            tag_contents[0], block_subtype
+                        )
+                    if block_type == "templated" and tag_contents:
+                        assert m_open and m_close
+                        raw_slice_info = self.track_templated(
+                            m_open, m_close, tag_contents
+                        )
+                self.update_inside_set_or_macro(block_type, tag_contents)
+                m_strip_right = regex.search(
+                    r"\s+$", raw, regex.MULTILINE | regex.DOTALL
+                )
+                if raw.startswith("-") and m_strip_right:
                     # Right whitespace was stripped. Split off the trailing
                     # whitespace into a separate slice. The desired behavior is
```

Comment:
At this point, we're inside this `if` block:
```
if elem_type.endswith("_end") or elem_type == "raw_begin":
```

To confirm this, I set a breakpoint in the tests, and when I hit it, I saw the following:
```
> /Users/bhart/dev/sqlfluff/src/sqlfluff/core/templaters/slicers/tracer.py(316)analyze()
-> pass
(Pdb) print(raw)
-}}
  
(Pdb) print(self.raw_str)
SELECT
  {{ 'col1,' -}}
  col2

```

Reply:
Yeah it's the `or elem_type == "raw_begin":` that concerned me.

Maybe add another `if elem_type.endswith("_end")` to this `if statement`?

```suggestion
                if elem_type.endswith("_end") and raw.startswith("-") and m_strip_right:
                    # Right whitespace was stripped after closing block. Split off the trailing
                    # whitespace into a separate slice. The desired behavior is
```

Reply:
I'll make that change. In the future, maybe there's a way to break apart this section of code. It's a bit trickier to reason about since it mixes handling of two different cases (block end and "raw").